### ISSUE TITLE
Why should we inherit from : IDynamicMetaObjectProvider

### ISSUE THREAD

**Issue by dineshtsm** [AntyaDev/KingAOP#14]
Why are we inheriting from : IDynamicMetaObjectProvider and the object should be type of dynamic. Is it not possible to make it work without changing the object to dynamic type  and inheriting from the above Interface ?

**Comment by tonyhallett** [comment]
No.  If you do that the class member will be used without going through the interface.  There will be no interception and the expression trees that invoke your aspects and the original code will not be generated.
